mne_bids/write.py

```diff
@@ -1881,6 +1851,30 @@ def write_raw_bids(raw, bids_path, events_data=None, event_id=None,
                              f'Please use one of {CONVERT_FORMATS[datatype]} '
                              f'for {datatype} datatype.')
 
+    # raise error when trying to copy files (copyfile_*) into same location
+    # (src == dest, see https://github.com/mne-tools/mne-bids/issues/867)
+    if bids_path.fpath.exists() and not convert and \
+            bids_path.fpath.as_posix() == Path(raw_fname).as_posix():
+        raise RuntimeError(f'Desired output BIDSPath ("{bids_path.fpath}") is'
```

Comment:
Should be a `FileExistsError`

I would also suggest to change indentation to the "black" style